google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ConnectionStatementExecutorImpl.java

```diff
@@ -442,4 +450,112 @@ public StatementResult statementShowRPCPriority() {
   public StatementResult statementShowTransactionIsolationLevel() {
     return resultSet("transaction_isolation", "serializable", SHOW_TRANSACTION_ISOLATION_LEVEL);
   }
+
+  private StatementResult getStatementResultFromQueryPlan(QueryPlan qp){
+
+    if(qp == null) return null;
+
+    ArrayList<Struct> list = new ArrayList<>();
+
+    for(PlanNode pn : qp.getPlanNodesList()){
+      list.add(Struct.newBuilder().set("QUERY PLAN").to(pn.getDisplayName()).build());
+
+    }
+
+    ResultSet rs = ResultSets.forRows(
+        Type.struct(StructField.of("QUERY PLAN", Type.string()))
+        , list);
+
+
+    return StatementResultImpl.of(rs);
+  }
+
+  private StatementResult executeStatement(String sql, QueryAnalyzeMode qam){
+    Statement statement = Statement.newBuilder(sql).build();
+    ResultSet rs = getConnection().analyzeQuery(statement, qam);
+    if(rs.getStats() != null){
+      return getStatementResultFromQueryPlan(rs.getStats().getQueryPlan());
+    }
+    throw SpannerExceptionFactory.newSpannerException(ErrorCode.UNAVAILABLE, String.format("Couldn't fetch stats for %s",sql));
+
+  }
+
+  /*
+  * This method supports the EXPLAIN feature with ANALYSE option only.
+  * Any option other than ANALYSE is not supported currently.
+  * If the statement has combination of analyse with any other option
+  * like "explain (analyse, costs) select * from table"
+  * it will throw exception .
+  */
+
+  @Override
+  public StatementResult statementExplain(String sql) {
+
+    if(sql == null){
+      throw SpannerExceptionFactory.newSpannerException(
+          ErrorCode.INVALID_ARGUMENT, String.format("Invalid String with Explain"));
+
+    }
+
+    if(sql.charAt(0) == '('){
```

Comment:
A SQL statement could in theory start with a '('. If you try to execute `explain (select * from all_types limit 1);` on a PostgreSQL database, it will return the query plan of the query between the parens.